packages/notebook/src/browser/service/notebook-monaco-text-model-service.ts

```diff
@@ -0,0 +1,45 @@
+// *****************************************************************************
+// Copyright (C) 2024 TypeFox and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { ReferenceCollection, URI, Reference, Event } from '@theia/core';
+import { injectable } from '@theia/core/shared/inversify';
+import { MonacoTextModelService } from '@theia/monaco/lib/browser/monaco-text-model-service';
+import { MonacoEditorModel } from '@theia/monaco/lib/browser/monaco-editor-model';
+import { NotebookModel } from '../view-model/notebook-model';
+
+/**
+ * special service for creating monaco textmodels for notebook cells.
+ * Its for optimization purposes since there is alot of overhead otherwise with calling the backend to create a document for each cell and other smaller things.
+ */
+@injectable()
+export class NotebookMonacoTextModelService extends MonacoTextModelService {
```

Comment:
Can you give a reason why this class extends (and replaces via rebind) the global ITextModelService? The functionality here does not change the behavior of the extended class in any way, as far as I can see.

Reply:
What I mean is this: the "document" as far as the VS Code API is concerned seems to be the main notebook document. Are these "mini-documents" not just local, ephemeral structures that the back-end does not need to know about?

Reply:
thanks alot for taking the time. Im not super happy with this solution but don't really know how to make it better.
To eleborate a bit more on this. The jupyter extension seem to require the documents in the backend to read and change cell content as well as to get the code to execute. The source paramter is only changed when saving and reading the document initially. (actually i just found two possible problems there). 
So as described above i need the functionality from the MonacoTextModelService to create the documents on fronted side but without it calling the plugin api to create the documents on backend side because i do that when i create the notebook document in the backend. If i would not do that it would make a call for each cell which takes quite some time. Calling the plugin API is done through a listener being called from the reference collection thats why i have a seperate reference collection for these notebook cell models. So this was the best way i could think of to reuse the functionality. If I would create a second i think there would be some issues with notifying about changes

Reply:
Fair enough that you need the cell documents in the back end, but still, this class neither calls nor overrides anything from the base class. Why is it not a separate, unrelated class?

Reply:
th `loadModel()` call